deps/rabbit/src/rabbit_variable_queue.erl

```diff
@@ -1886,19 +2255,27 @@ publish_delivered1(Msg = #basic_message { is_persistent = IsPersistent,
                                  needs_confirming = NeedsConfirming },
                    _ChPid, _Flow, PersistFun,
                    State = #vqstate { mode                = lazy,
+                                      version             = Version,
                                       qi_embed_msgs_below = IndexMaxSize,
                                       next_seq_id         = SeqId,
+                                      next_deliver_seq_id = NextDeliverSeqId,
                                       out_counter         = OutCount,
                                       in_counter          = InCount,
                                       durable             = IsDurable,
                                       unconfirmed         = UC }) ->
     IsPersistent1 = IsDurable andalso IsPersistent,
-    MsgStatus = msg_status(IsPersistent1, true, SeqId, Msg, MsgProps, IndexMaxSize),
+    MsgStatus = msg_status(Version, IsPersistent1, true, SeqId, Msg, MsgProps, IndexMaxSize),
     {MsgStatus1, State1} = PersistFun(true, true, MsgStatus, State),
     State2 = record_pending_ack(m(MsgStatus1), State1),
     UC1 = gb_sets_maybe_insert(NeedsConfirming, MsgId, UC),
     State3 = stats({0, 1}, {none, MsgStatus1}, 0,
                    State2 #vqstate { next_seq_id      = SeqId    + 1,
+                                     %% @todo Update next_deliver_seq_id
+                                     %% @todo Assert when doing the update to be sure there are no "deliver" holes.
```

Comment:
I'm not sure what I meant by this. Spend some time to figure this out, this sounds important.